mlflow/utils/search_utils.py

```diff
@@ -187,7 +198,21 @@ def _invalid_statement_token(cls, token):
     def _process_statement(cls, statement):
         # check validity
         invalids = list(filter(cls._invalid_statement_token, statement.tokens))
-        if len(invalids) > 0:
+        if len(invalids) % 3 == 0 and len(invalids) != 0:
+            groupby_invalids = [invalids[i:i+3] for i in range(0, len(invalids), 3)]
+            invalid_clauses = ""
+            for invalid in groupby_invalids:
+                if invalid[1].value.lower() in ['like', 'ilike']:
+                    like_comparison = Comparison(TokenList(invalid))
+                    statement.tokens = [x for x in statement.tokens if type(x) == Comparison]
+                    statement.tokens.append(like_comparison)
```

Comment:
Can parsing and validating `LIKE` and `ILIKE` tokens be pushed down to `_invalid_statement_token`. Let's keep the logic of on invalids as purely invalid.

Reply:
I can do that, but it would be even better to wait for this pr from @dbczumar to be merged and released: https://github.com/andialbrecht/sqlparse/pull/525
What do you think ? Do I make the change while waiting for the pr to be merged ?

Reply:
I posted an issue asking when they plan to release @dbczumar's change: https://github.com/andialbrecht/sqlparse/issues/533